src/renderer/captureTool/js/capture.js

```diff
@@ -0,0 +1,1121 @@
+                return item;
+            }
+        });
+        if (checkVal) {
+            fluid.each(solutionData.settings, function (item, idx) {
+                if (!fluid.contains(curSettings, solutionId + ":" + idx)) {
+                    curSettings.push(solutionId + ":" + idx);
+                }
+            });
+        }
+        else {
+            fluid.each(solutionData.settings, function (item, idx) {
+                fluid.remove_if(curSettings, function (curSetting) {
+                    return curSetting === solutionId + ":" + idx;
+                });
+            });
+        }
+        that.applier.change("settingsToKeep", curSettings);
+    };
+
+    gpii.captureTool.onHideShowNumSettingLink = function (currentTarget) {
+        var solutionId = currentTarget.dataset.solution;
+        var button = $(".flc-hideshow-solution[data-solution='" + solutionId + "']");
+        button.click();
+    };
+
+    gpii.captureTool.onHideShowSolution = function (that, currentTarget) {
+        var solutionId = currentTarget.dataset.solution;
+        var hideShow = currentTarget.dataset.show;
+        if (hideShow === true || hideShow === "true") {
+            $("#flc-settings-for-" + solutionId.replace(/\./g, "\\.")).toggle(hideShow);
+            currentTarget.dataset.show = false;
+            currentTarget.value = "Show Settings";
+        }
+        else {
+            $("#flc-settings-for-" + solutionId.replace(/\./g, "\\.")).toggle(hideShow);
+            currentTarget.dataset.show = true;
+            currentTarget.value = "Hide Settings";
+        }
+
+    };
+
+    gpii.captureTool.onSelectPrefssetDropdown = function (that, currentTarget) {
+        var curSelected = that.locate("currentlySelectedPrefset");
+        curSelected[0].innerHTML = currentTarget.innerHTML;
+        var curTarget = $(currentTarget);
+        that.applier.change("prefsSetSaveType", curTarget.data("type"));
+        that.applier.change("selectedPrefsSet", curTarget.data("prefsset-id"));
+        that.applier.change("selectedPrefsSetName", curTarget.data("prefsset-name"));
+    };
+
+    gpii.captureTool.calculateCurrentNumberOfTotalSettingsToRender = function (capturedSettingsToRender) {
+        var tally = 0;
+        fluid.each(capturedSettingsToRender, function (item) {
+            tally += item.numberOfRenderSettings;
+        });
+        return tally;
+    };
+
+    gpii.captureTool.updateCurrentNumberOfTotalSettingsToRender = function (applierFunc, capturedSettingsToRender) {
+        var tally = gpii.captureTool.calculateCurrentNumberOfTotalSettingsToRender(capturedSettingsToRender);
+        applierFunc("currentNumberOfTotalSettingsToRender", tally);
+    };
+
+    gpii.captureTool.updateCapturedSettingsToRender = function (that, showDefaultSettings) {
+        var togo = {};
+        fluid.each(that.model.capturedSettings, function (appData, appId) {
+            if (appData.numberOfSettings >= 0) { // sgithens DEMO_TOGGLE
```

Comment:
What is DEMO_TOGGLE ?

Reply:
Oh, this was an old comment from early the UI development where we weren't sure if settings with zero settings should show up at all in the UI so I toggling this if statement. 

I've removed this comment.